NLP/main/extract_display_quotes.py

```diff
@@ -77,164 +80,87 @@ def __init__(self):
+            if file.lower().endswith('txt'):
+                text_dic = self.load_txt(self.file_uploader.value[file])
             else:
-                text_names.append(text_name)
-
-            try:
-                # read the text file
-                doc_lines = codecs.decode(txt_upload.value[input_file]['content'], encoding='utf-8')
-                
-                # store them inside a dictionary
-                text_dict['text_id'] = text_id
-                text_dict['text_name'] = text_name
-                text_dict['text'] = doc_lines
-                all_files.append(text_dict)
-                    
-            except:
-                # this will provide some information in the case of an error
-                app_logger.exception("message")
-                traceback.print_exc()
-        
-        # convert the extracted texts into a pandas dataframe for further processing
-        self.text_df = pd.DataFrame.from_dict(all_files)
-        self.text_df['spacy_text'] = self.text_df['text'].apply(lambda text: self.nlp_preprocess(text))
-        self.text_df.set_index('text_id', inplace=True)
-        
-        return self.text_df
-    
-    
-    def process_xls(self, xls_upload):
-        '''
-        Pre-process uploaded .xlsx file into pandas dataframe
+                text_dic = self.load_table(self.file_uploader.value[file], \
+                    file_fmt=file.lower().split('.')[-1])
+            all_data.extend(text_dic)
+        uploaded_df = pd.DataFrame.from_dict(all_data)
 
-        Args:
-            xls_upload: the uploaded .xlsx file from upload_files()
-        '''
-        text_ids = []
-        
-        # read the excel file containing the list of texts and convert them into a pandas dataframe
-        try:
-            self.text_df = pd.read_excel(io.BytesIO(xls_upload.data[0]))
-        except:
-            self.text_df = pd.read_csv(io.BytesIO(xls_upload.data[0]))
-        
-        # generate dated unique identifier for text_id
-        for row in self.text_df.itertuples():
-            
-            text_id = datetime.now().strftime('%Y-%m-%d-') + str(uuid4())[:10]
-            while text_id in text_ids:
-                text_id = datetime.now().strftime('%Y-%m-%d-') + str(uuid4())[:10]
-            else:
-                text_ids.append(text_id)
-        self.text_df['text_id'] = text_ids
-        self.text_df.set_index('text_id', inplace=True)
-        
-        # process text using spaCy
-        self.text_df['spacy_text'] = self.text_df['text'].apply(lambda text: self.nlp_preprocess(text))
-        
-        return self.text_df
+        uploaded_df = self.hash_gen(uploaded_df)
+        uploaded_df = self.nlp_preprocess(uploaded_df)
+        self.text_df = pd.concat([self.text_df, uploaded_df])
+        # Deduplicate the text_df by text_id
+        if deduplication:
+            self.text_df.drop_duplicates(subset='text_id', keep='first', inplace=True)
```

Comment:
user can no longer see text_df from the notebook. I'd think it'd be useful for them to check what's been uploaded

Reply:
also need to check save preview and show top entities code - not working properly at
<img width="847" alt="save-preview" src="https://user-images.githubusercontent.com/61028906/177924047-10ad5841-7deb-4326-a23c-80626cb99048.png">
m

Reply:
Can't people access qt.text_df? I think that worked for me. I haven't touched the save preview/top entities codes, but possibly some dataframe structure has changed. Most likely because I am not making the text_id column as index because it's causing problem when I concat new df to the self.text_df.